### PR TITLE
Add functions for internal types

### LINKED ISSUE #29: Use factory function for all internal types.
Been caught out by uninitialized fields too often.

### PR DESCRIPTION
**Pull request by koden-km**
Fixes #29

I'm not sure if all these were needed. Let me know and i'll update the changes.

### CODE REVIEW

**Comment by jmalloc** [comment]
FYI @koden-km, I messed up your PR when I merged #126 -- we can go through these some time next week if you like, or I can cherry pick them out if you can't be bothered rebasing.

**Comment by koden-km** [comment]
I have updated from master and fixed the merge conflicts.

File: src/rinq/internal/localsession/revision.go [hunk 1]
```diff
@@ -15,6 +15,20 @@ type revision struct {
 	logger  rinq.Logger
 }
 
+func newRevision(
```
Comment (jmalloc) [inline src/rinq/internal/localsession/revision.go]: As discussed, this should probs just be "positional-inline-construction".

File: src/rinq/internal/remotesession/revision.go [hunk 2]
```diff
@@ -12,6 +12,13 @@ type revision struct {
 	catalog *catalog
 }
 
+func newRevision(ref ident.Ref, cat *catalog) rinq.Revision {
```
Comment (jmalloc) [inline src/rinq/internal/remotesession/revision.go]: Another positional one.

**Review by jmalloc — CHANGES_REQUESTED** [review]

**Review by jmalloc — APPROVED** [review]

**Pull request merged**